core/app/models/spree/inventory_unit.rb

```diff
@@ -155,5 +221,11 @@ def ensure_can_destroy
         throw :abort
       end
     end
+
+    def change_state!(new_state)
+      previous_state = state
```

Comment:
Can we try to make use of `ActiveModel::Dirty` here?

http://api.rubyonrails.org/classes/ActiveModel/Dirty.html